lib/modules/datasource/helm/common.ts

```diff
@@ -1,14 +1,30 @@
+import { parseGitUrl } from '../../../util/git/url';
 import { regEx } from '../../../util/regex';
 import type { HelmRelease, RepoSource } from './types';
 
 const chartRepo = regEx(/charts?|helm|helm-charts/i);
-const githubUrl = regEx(
-  /^(?<url>https:\/\/github\.com\/[^/]+\/(?<repo>[^/]+))(:?\/|\/tree\/[^/]+\/(?<path>.+))?$/
-);
 const githubRelease = regEx(
   /^(https:\/\/github\.com\/[^/]+\/[^/]+)\/releases\//
 );
 
+function splitRepoUrl(url: string): RepoSource | null {
+  const parsed = parseGitUrl(url);
+  if (!chartRepo.test(parsed.name)) {
+    return null;
+  }
+
+  if (parsed.filepathtype === 'tree') {
+    return {
+      sourceUrl: parsed.toString(),
+      sourceDirectory: parsed.filepath,
+    };
+  }
+
+  return {
+    sourceUrl: parsed.toString(),
+  };
```

Comment:
i still think this split should be done here instead:

https://github.com/renovatebot/renovate/blob/0fd190edc9760994057ca9d1cb5b46022c23d3d3/lib/modules/datasource/metadata.ts#L115

Reply:
Sure, and thanks a ton for the hint on where to implement it!

I've revised my implementation accordingly, including the removal of splitting in npm since we'll have that in a central place.